File: ZezeJava/ZezeJava/src/main/java/Zeze/Transaction/Collections/LogMap.java
package Zeze.Transaction.Collections;

import Zeze.Transaction.Bean;
import Zeze.Transaction.Changes;
import Zeze.Transaction.Log;
import org.jetbrains.annotations.NotNull;

public abstract class LogMap<K, V> extends LogBean {
	private @NotNull org.pcollections.PMap<K, V> value;

	protected LogMap(Bean belong, int varId, Bean self, @NotNull org.pcollections.PMap<K, V> value) {
		super(belong, varId, self);
		this.value = value;
	}

	@Override
	public abstract int getTypeId();

	public final @NotNull org.pcollections.PMap<K, V> getValue() {
		return value;
	}

	final void setValue(@NotNull org.pcollections.PMap<K, V> value) {
		this.value = value;
	}

	@Override
	public void collect(@NotNull Changes changes, @NotNull Bean recent, @NotNull Log vlog) {
		throw new UnsupportedOperationException("Collect Not Implement.");
	}

	@SuppressWarnings("unchecked")
	@Override
	public void commit() {
		((PMap<K, V>)getThis()).map = value;
	}
}

File: Zege/Zeze/Transaction/Collections/LogMap.cs
using System;
using System.Collections.Generic;
using System.Collections.Immutable;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Zeze.Serialize;

namespace Zeze.Transaction.Collections
{
	public abstract class LogMap<K, V> : LogBean
	{
#if !USE_CONFCS
		internal ImmutableDictionary<K, V> Value { get; set; }

		public override void Collect(Changes changes, Bean recent, Log vlog)
		{
			throw new NotImplementedException($"Collect Not Implement.");
		}

		public override void Commit()
		{
			((CollMap<K, V>)This)._map = Value;
		}
#endif
	}
}
